go-controller/pkg/ovn/gateway_init.go

```diff
@@ -523,6 +530,15 @@ func (oc *Controller) addNodeLocalNatEntries(node *kapi.Node, mgmtPortMAC string
 		}()
 	}
 
+	if annotationPresent {
+		stdout, stderr, err := util.RunOVNNbctl("--if-exists", "lr-nat-del", ovnClusterRouter, "dnat_and_snat",
+			externalIP.String())
+		if err != nil {
+			return fmt.Errorf("failed to delete dnat_and_snat entry for the management port on node %s, "+
+				"stdout: %s, stderr: %q, error: %v", node.Name, stdout, stderr, err)
+		}
+	}
+
 	mgmtPortName := util.K8sPrefix + node.Name
```

Comment:
Should we just call `lr-nat-del` regardless? Keeping in mind that we do the same in other cases. If we do that you can do lr-nat-del and lr-nat-add in one transaction like other cases?

Imagine this case, the K8s node got deleted and was added back. During that time, we failed to remove this entry from OVN DB. Now on the node addition, annotation is not set on the node and it gets a new IP. This new iP will be different from what is in the OVN DB and it will error out.

Reply:
Sure. I can add the lr-nat-del check outside and do it in a single transaction.